src/lightning/data/streaming/data_processor.py

```diff
@@ -704,6 +737,8 @@ def __init__(
             delete_cached_files: Whether to delete the cached files.
             fast_dev_run: Whether to run a quick dev run.
             random_seed: The random seed to be set before shuffling the data.
+            reorder_files: By default, reorders the files by file size to distribute work equally among all workers.
+                Set this to ``False`` if the order in which samples are processed should be preserved.
```

Comment:
@lantiga @tchaton 
Added this arg to opt-out of the optimization based on your suggestion. Is this name ok or are there better suggestions?

Reply:
Looks great ! Let's land this.